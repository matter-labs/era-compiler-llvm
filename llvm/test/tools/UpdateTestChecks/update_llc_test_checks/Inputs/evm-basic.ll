; RUN: llc < %s -mtriple=evm | FileCheck %s

define i256 @swap_second_no_junk(i256 %a1, i256 %a2, i256 %a3, i256 %a4) nounwind {
  %x1 = sub i256 %a4, %a1
  ret i256 %x1
}
