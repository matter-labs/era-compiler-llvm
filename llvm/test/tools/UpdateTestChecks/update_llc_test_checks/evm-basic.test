# REQUIRES: evm-registered-target

# RUN: cp -f %S/Inputs/evm-basic.ll %t.ll && %update_llc_test_checks %t.ll
# RUN: diff -u %S/Inputs/evm-basic.ll.expected %t.ll
