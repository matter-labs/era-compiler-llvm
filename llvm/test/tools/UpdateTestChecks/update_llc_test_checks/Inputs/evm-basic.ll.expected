; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 2
; RUN: llc < %s -mtriple=evm | FileCheck %s

define i256 @swap_second_no_junk(i256 %a1, i256 %a2, i256 %a3, i256 %a4) nounwind {
; CHECK-LABEL: swap_second_no_junk:
; CHECK:       ; %bb.0:
; CHECK-NEXT:    JUMPDEST
; CHECK-NEXT:    SWAP3
; CHECK-NEXT:    SWAP2
; CHECK-NEXT:    POP
; CHECK-NEXT:    POP
; CHECK-NEXT:    SUB
; CHECK-NEXT:    SWAP1
; CHECK-NEXT:    JUMP
  %x1 = sub i256 %a4, %a1
  ret i256 %x1
}
